openedx/core/djangoapps/content_libraries/rest_api/serializers.py

```diff
@@ -357,9 +359,42 @@ class ContentLibraryComponentKeysSerializer(serializers.Serializer):
     usage_keys = serializers.ListField(child=UsageKeyV2Serializer(), allow_empty=False)
 
 
-class ContentLibraryComponentCollectionsUpdateSerializer(serializers.Serializer):
+class OpaqueKeySerializer(serializers.BaseSerializer):
     """
-    Serializer for adding/removing Collections to/from a Component.
+    Serializes a OpaqueKey with the correct class.
+    """
+    def to_representation(self, value: OpaqueKey) -> str:
+        """
+        Returns the OpaqueKey value as a string.
+        """
+        return str(value)
+
+    def to_internal_value(self, value: str) -> OpaqueKey:
+        """
+        Returns a UsageKeyV2 or a LibraryContainerLocator from the string value.
+
+        Raises ValidationError if invalid UsageKeyV2 or LibraryContainerLocator.
+        """
+        try:
+            return UsageKeyV2.from_string(value)
+        except InvalidKeyError:
+            try:
+                return LibraryContainerLocator.from_string(value)
+            except InvalidKeyError as err:
+                raise ValidationError from err
```

Comment:
If we genuinely don't care what kind of OpaqueKey we deserialize here, this can be simpler:
```suggestion
    def to_internal_value(self, value: str) -> OpaqueKey:
        """
        Returns an OpaqueKey from the string value.

        Raises ValidationError if invalid OpaqueKey.
        """
        try:
            return OpaqueKey.from_string(value)
        except InvalidKeyError as err:
            raise ValidationError from err
```

Reply:
We pass it through functions as OpaqueKeys, but we have some checks on the keys to check its type:
https://github.com/open-craft/edx-platform/blob/c712133864d3069cc160a6e2f41d04101fc60e30/openedx/core/djangoapps/content_libraries/api/collections.py#L132-L159

Reply:
That's cool -- then we probably don't need to duplicate these checks in the serializer too, right?